### PR TITLE
Feature/dvop 2567

### PR DESCRIPTION
**Pull request by LJMRBUMP**
- EFS deployed & working
- Newly deployed instances auto-mount
- Instances can access directories & files created by existing instances
- Instances can read & write files
- Screenshots of mount points & test files can be found in:

[DVOP-2567](https://companieshouse.atlassian.net/browse/DVOP-2567)

[DVOP-2567]: https://companieshouse.atlassian.net/browse/DVOP-2567?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

UPDATE:
All recommended defaults have been crossed checked and are being applied including noresvport.

TLS now enabled, EFS system policy amended. Efs-mount-watchdog now enabled on the tunnel

### CODE REVIEW

File: groups/instance/data.tf [hunk 1]
```diff
@@ -70,7 +70,7 @@ data "aws_ami" "artifactory_ami" {
   most_recent = true
   owners      = [local.ami_owner_id]
   name_regex  = "${var.service}-${var.default_ami_version_pattern}"
-
+
```
Comment (sienkin) [inline groups/instance/data.tf:73]: This should just be an empty line

File: groups/instance/cloud-init/templates/enable_config.tpl [hunk 2]
```diff
@@ -808,5 +764,8 @@ write_files:
       ${artifactory_license}
 
 runcmd:
-  - systemctl enable artifactory 
-  - systemctl restart artifactory
\ No newline at end of file
+  - systemctl enable artifactory
+  - sudo mkdir /var/lib/artifactory
+  - sudo mount -t efs ${efs_dns_name}:/ /var/lib/artifactory
+  - sudo echo "${efs_dns_name}:/ /var/lib/artifactory efs defaults,_netdev 0 0" >> /etc/fstab
```
Comment (sienkin) [inline groups/instance/cloud-init/templates/enable_config.tpl:770]: May be worth reviewing the mount options here to ensure we're setting sensible defaults for NFS/EFS

E.g:
https://docs.aws.amazon.com/efs/latest/ug/mounting-fs-nfs-mount-settings.html

File: groups/instance/data.tf (hunk 1 above)
Comment (LJMRBUMP) [reply groups/instance/data.tf]: Change implemented via: [
](https://github.com/companieshouse/artifactory-terraform/pull/20/commits/b658eab3c8890b731667e25b692d3da0e9851d06)

File: groups/instance/cloud-init/templates/enable_config.tpl (hunk 2 above)
Comment (LJMRBUMP) [reply groups/instance/cloud-init/templates/enable_config.tpl]: Hi Wayne, I have crossed checked the logs of what values are being implemented from the mounting commands where these match the defaulted recommendations. Although ....

I did notice that I was not using TLS with either of the EFS mounting commands.

To enable TLS, I have changed the EFS system policy to set the ‘secureTransport’ to true. From there I have added the ‘tls’ option to both mount commands. This also results in the efs-mount-watchdog being enabled & is monitoring the TLS tunnel config for the EFS & Instance. 

Tested with a new deployment, reboot & stop / start process of the instance. EFS continues to connect with read & write abilities remaining.

**Review by sienkin — APPROVED** [review]

**Pull request merged**